src/util/models.py

```diff
@@ -573,7 +573,34 @@ def remove_inherit_from_model(cr, model, inherit, keep=(), skip_inherit=(), with
             table = table_of_model(cr, relation)
             irs = [ir for ir in indirect_references(cr) if ir.table == table and ir.res_id is not None]
             for ir in irs:
-                query = 'DELETE FROM "{}" WHERE ({})'.format(ir.table, ir.model_filter())
+                if not ir.company_dependent_comodel:
+                    query = format_query(cr, "DELETE FROM {} WHERE ({})", ir.table, ir.model_filter())
+                elif ir.company_dependent_comodel:
+                    cr.execute("SELECT id FROM ir_model WHERE model = %s", [model])
+                    [mod_id] = cr.fetchone() or [None]
+                    if mod_id:
+                        query = cr.mogrify(
+                            format_query(
+                                cr,
+                                """
+                                UPDATE {table}
+                                   SET {field} = (
+                                       SELECT jsonb_object_agg(
+                                              key,
+                                              CASE
+                                                 WHEN value::int4 = %s THEN NULL
+                                                 ELSE value::int4
+                                              END)
+                                         FROM jsonb_each_text({field})
+                                        )
+                                 WHERE {field} IS NOT NULL
+                                   AND {field} @? %s
+                                """,
+                                table=ir.table,
+                                field=ir.res_id,
+                            ),
+                            [mod_id, "$.* ? (@ == {})".format(mod_id)],
+                        )
```

Comment:
This is wrong.

You copy/paste this part from `util.remove_model`, which is specific for the removal of one2many fields pointing to `ir.model` records.

Actually, we don't need this part at all.
We have 3 possible case:
1. `ir.company_dependent_comodel == model`: the one2many explicitly point to the model, which is really weird, as the many2one we remove come from the inherit model. Anyway, there are no records to remove here.
2. `ir.company_dependent_comodel in inherit_models`: the inherits models (and records) are still there; nothing to do.
3. `else`: unrelated field; skip.

```suggestion
```